### ISSUE TITLE
IMobileServiceTableOperation.ExecuteAsync hides actual error information

### ISSUE THREAD

**Issue by kinex** [Azure/azure-mobile-apps-net-client#298]
Here is typical code in my TableController:

```
public class EventController : TableController<Event>
{
        public async Task<Event> PatchEvent(string id, Delta<Event> patch)
        {
            var current = Lookup(id);
            if (current == null)
            {
                // problem: error information is not delivered to IMobileServiceSyncHandler
                throw new HttpResponseException(this.Request.CreateNotFoundResponse());
            }

            var userId = await IdentityHelper.FindSidAsync(User, Request).ConfigureAwait(false);

            if (current?.Queryable?.FirstOrDefault()?.UserId == userId)
            {
                return await UpdateAsync(id, patch).ConfigureAwait(false);
            }
            else
            {
                return null;
            }
        }
...
}

```

Then on the client side I am trying to handle sync conflicts as follows:
```

    internal class MobileServiceSyncHandler : IMobileServiceSyncHandler
    {
        public async Task<JObject> ExecuteTableOperationAsync(IMobileServiceTableOperation operation)
        {
            MobileServiceInvalidOperationException error = null;
            Func<Task<JObject>> tryExecuteAsync = operation.ExecuteAsync;

            int retries = 0;
            do
            {
                try
                {
                    error = null;
                    var result = await tryExecuteAsync().ConfigureAwait(false);
                    return result;
                }
                catch (MobileServiceConflictException e)
                {
                    error = e;
                }
                catch (MobileServicePreconditionFailedException e)
                {
                    error = e;
                }
                catch (MobileServiceInvalidOperationException e)
                {
                    error = e;
                }
                catch (Exception e)
                {
                    throw e;
                }

                if (error != null)
                {
                    // problem: StatusCode is never "NotFound"
                    if (error.Response?.StatusCode == System.Net.HttpStatusCode.NotFound && operation.Kind == MobileServiceTableOperationKind.Update)
                    {
                        // tried to update an item that does not exists on server, change to insert operation
                        tryExecuteAsync = async () => await operation.Table.InsertAsync(operation.Item).ConfigureAwait(false) as JObject;
                        continue;
                    }
                    else
                    {
						...
                    }
                }
            }
            while (error != null && ++retries < 2);

            if (error != null)
            {
                operation.AbortPush();
            }

            return null;
        }
    }

```
So if I throw "HttpResponseException(this.Request.CreateNotFoundResponse())" in my TableController, the exception thrown from  IMobileServiceTableOperation.ExecuteAsync is "MobileServiceInvalidOperationException" with a message "Mobile Service table operation returned an unexpected response." and there is no other useful information available from the exception object. I guess MobileServiceInvalidOperationException may be thrown for several other reasons too, so I cannot assume it is always the "NotFound" case.

In other words it is now very difficult to handle a conflict where we are trying to update an item that does not exist on the server.

**Comment by adrianhall** [comment]
There is an embedded exception in MobileServiceInvalidOperationException.  Have you checked that?

**Comment by kinex** [comment]
Yes I checked everything that is passed to the method. If I remember correctly InnerException is null.

**Comment by elamalani** [comment]
Thanks for using Azure Mobile Apps. I'm closing this question as it seems to be over a year old. Please feel free to reach out to us in case you have any questions.
